File: C#/021.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ProjectEuler.Problems
{
	class Problem021 : IProblem
	{

		public void CalculateSolution()
		{
			const int limit = 10000;
			Func<int, int> d = (n => Helper.Methods.GetFactors(n)
			                               .Sum() - n);

			var sum = 0;

			for (var i = 1; i <= limit; i++)
			{
				var divsum = d(i);
				var complement = d(divsum);

				if (i == complement && i != divsum)
					sum += i;	//the complement will be added later;
			}

			this.Solution = sum.ToString();
		}


		public string Solution { get; private set; }

		public string Hint { get; private set; }

	}
}

File: Python/021.py
from mathlib import divsum

dsums = {n: divsum(n) for n in range(2, 10000)}

sum = 0
for n, s in dsums.items():
	if (n == s or s == 1): continue
	if (s < len(dsums) and dsums[s] == n):
		sum += n
		
print(sum)